addon/components/bs-form.d.ts

```diff
@@ -0,0 +1,40 @@
+import BsButton from './bs-button';
+import BsFormElement from './bs-form/element';
+
+import Component from '@glimmer/component';
+
+interface BsFormSignature<T> {
+  Element: HTMLDivElement;
+  Args: {
+    disabled?: boolean;
+    formLayout?: 'vertical' | 'horizontal' | 'inline';
+    hideValidationsOnSubmit?: boolean;
+    horizontalLabelGridClass?: string;
+    model?: T;
+    preventConcurrency?: boolean;
+    readonly?: boolean;
+    submitOnEnter?: boolean;
+    onBefore?: (model: T) => void;
+    onInvalid?: (model: T, error: unknown) => void;
+    onSubmit?: (model: T, result: unknown) => void;
```

Comment:
Consumer can return a promise to control submission state of the form.

```suggestion
    onSubmit?: (model: T, result: unknown) => Promise<void> | void;
```